python/tests/test_highlevel.py

```diff
@@ -754,6 +754,7 @@ def test_compute_mutation_time(self):
                 np.allclose(python_time, tables.mutations.time, rtol=1e-15, atol=1e-15)
             )
             # Check we have valid times
+            tables.sort()
```

Comment:
Why do we need to sort here? This would be a bug in compute_mutation_times, wouldn't it?

Reply:
This is failing on the example tree from `insert_branch_mutations` looking into it.

Reply:
Massive confusion till I realised this test uses a non-fixed random seed! :facepalm:

Reply:
Dammit, I hate it when that happens! :smile:

Reply:
Here's the badly sorted data that's come out of `compute_mutation_times`. Just a single site. I'm trying to get my head round what's going on. https://gist.github.com/benjeffery/fe7727ae93bbe79fa5b3509428097465

Reply:
I don't think this is a bug in `compute_mutation_times` The order going in is just "sites and parents before children" but once times are applied (using the node times) the old order is no longer valid.

Reply:
So I think it is a bug in compute_mutation_times if it doesn't return a well-formed mutation table. If we have to, we could do a site-wise sort, once all the times have been generated? (Or if this is too awkward, we could do a full TableCollection.sort(), but this is a very heavy weapon.)

Reply:
In https://github.com/tskit-dev/tskit/pull/726/commits/30a3134d8c6ff5258fec539d5f260dd99361f503 I've used `tsk_table_collection_sort` but only if it is needed. If this turns out to be a bottleneck we can put more effort in, but I'm not sure it is worth the complexity without a pressing use case.

Reply:
Need to change this to use a bookmark pointing to the end of the edge table to skip edge sorting.

Reply:
Skip edge sorting in https://github.com/tskit-dev/tskit/pull/726/commits/36ce487bba0a3b9aa45be0f013840ab690f1a21b